conninfo/src/conninfo-v1.0-wd01.md

```diff
@@ -0,0 +1,238 @@
+`connection-info`. The value of the entry MUST be of type `fields`.
+
+The following key values have defined meaning. An implementation MAY choose to omit a key from the map.
+An implementation MAY also include keys of its own. If an implementation receives a key from its partner
+that it does not recognise, it SHOULD ignore it.
+
+# 3.1 connection-info
+
+`<type name="connection-info" class="restricted" source="fields"/>`
+
+The content of `connection-info` allows the container to provide information about itself and its operating
+environment to its partner.
+
+**Table 1-1. connection-info**
+
+| Name | Description |
+| :--- | :--- |
+| `process-identifier` | The value of the process-identifier MUST be an AMQP string. The purpose of the value is to identify the process running this AMQP host. It is RECOMMENDED that this is an operating system PID. |
+| `network-host` | The value of network-host MUST be an AMQP string. The purpose of the value is to identify the host running the AMQP container in a way that assists those fault finding within a system. This value MAY be a hostname, fully qualified hostname or IP address. The value MAY be network resolvable by the AMQP peer but the peer MUST NOT rely on this. |
+| `amqp-product` | The value of amqp-product MUST be an AMQP string and is a `product-string`. It provides details of the software product(s) that provide the AMQP implementation. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the product. |
+| `product` | The value of product MUST be an AMQP string and is a product-string. It provides details of the software product(s) that is the end-user application. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the application. |
+| `platform` | The value of platform MUST be an AMQP string and is a product-string. It exists to expose details of a product’s runtime platform. This typically identify the operating system, Java/.NET runtimes etc. It MAY identify multiple products. By convention, the products are listed in order of their significance. |
+
+# 3.2 product-string
+
+`<type name="product-string" class="restricted" source="string"/>`
+
+The content of the `product-string` will adopt the content structure as the `User-Agent` header (defined by 
+[[RFC2616](#rfc2616)]) and commonly implemented by Web Browsing software. The field can contain multiple product tokens
+and comments identifying the product and any sub-products which form a significant part of the system. By convention, 
+the product tokens are listed in order of their significance
+
+Formally, the content of `product-string` MUST conform to this ABNF production rule:
+ 
+```
+product-string = 1*( product | comment ) 
+```
+
+where symbols `product`, `comment`, and the ABNF grammar itself are as defined by [[RFC2616](#rfc2616)].
```

Comment:
User-Agent is now defined RFC7231 https://tools.ietf.org/html/rfc7231#section-5.5.3

Reply:
I've updated to use newer RFC and replied avoid the redefinition completely.

Reply:
I am wondering about the type of `product-string` and whether string is the most appropriate choice in the AMQP ecosystem.     `product-string` comprises zero of more products each associated with zero or more comments.

A [composite type  _product_](http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#section-composite-type-representation) could be defined encapsulating a product name, version (optional), and zero or more comments.  the fields in the connection-info, `amqp-product` etc would then be defined as a list of these types.   However, taking this approach would harm interoperability (codec updates) and for something that is periphery concern, questionable value.

I wonder about a simpler approach - a list of lists 

```
[ [ "ApacheQpidJMS/0.36" ],
  [ "ApacheQpidProton/0.45", "build 12345", "another comment"]
]
```

the `car` of each inner list would always be a `product` (RFC7231).  the `crd` would be (zero or more) comment (RFC7231).

Reply:
My idea above was horrid. I have a slightly simpler suggestion on #8